src/unknown.rs

```diff
@@ -0,0 +1,50 @@
+use std::fs::File;
+use std::io;
+use std::path::Path;
+
+static ERROR_MESSAGE: &str = "Unknown Architecture";
+
+#[derive(Debug)]
+pub struct Handle;
```

Comment:
Could you add a comment on this type explaining the strategy here? i.e., This type is only used to make this compile on unsupported platforms, but otherwise always returns an error.